test/jdk/javax/swing/regtesthelpers/JRobot.java

```diff
@@ -242,10 +257,17 @@ public Point centerOf(Rectangle r, Point p) {
      * @param c a Component whose coordinate system is used for conversion
      */
     public void convertRectToScreen(Rectangle r, Component c) {
-        Point p = new Point(r.x, r.y);
-        SwingUtilities.convertPointToScreen(p, c);
-        r.x = p.x;
-        r.y = p.y;
+        AtomicReference<Point> p = new AtomicReference<>();
```

Comment:
So this change must be documented on all affected entry points, so that people understand
what it does.
Also touching Robot means you need to run ALL tests that use this utility and make sure you didn't break any. 
Have you done that ?

Reply:
Thanks @prrace for the reviews, 
I have added EDT support for 'getCenterPoint' and 'convertRectToScreen' functions, executed all the tests which are using this function and confirmed its working.

Reply:
But you seem to have ignored my comment about documenting it on 'all entry points' meaning EVERY exported API from this class which can end up here should say something like
"This method will invoke code on the Event Dispatch Thread".

Reply:
Sorry I didn't ignored your valuable comment regarding documentation.